src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +58,429 @@ class ServerFailed(Exception):
     """Server didn't start/stop properly."""
 
 
+class DaosServer(DaosCommand):
+    """Defines an object representing a server command."""
+
+    def __init__(self, path=""):
+        """Create a server command object
+
+        Args:
+            path (str): path to location of daos_server binary.
+        """
+        super(DaosServer, self).__init__(
+            "/run/daos_server/*", "daos_server", path)
+
+        self.yaml_params = self.DaosServerConfig()
+        self.timeout = 30
+        self.server_cnt = 1
+        self.mode = "normal"
+
+        self.debug = FormattedParameter("-b", None)
+        self.targets = FormattedParameter("-t {}")
+        self.config = FormattedParameter("-o {}")
+        self.port = FormattedParameter("-p {}")
+        self.storage = FormattedParameter("-s {}")
+        self.modules = FormattedParameter("-m {}")
+        self.xshelpernr = FormattedParameter("-x {}")
+        self.firstcore = FormattedParameter("-f {}")
+        self.group = FormattedParameter("-g {}")
+        self.attach = FormattedParameter("-a {}")
+        self.sock_dir = FormattedParameter("-d {}")
+        self.insecure = FormattedParameter("-i", None)
+
+    def get_params(self, test):
+        """Get params for Server object and server configuration."""
+        super(DaosServer, self).get_params(test)
+        self.yaml_params.get_params(test)
+
+    def set_config(self, yamlfile):
+        """Set the config value of the parameters in server command."""
+        self.config.value = self.yaml_params.create_yaml(yamlfile)
+        self.mode = "normal"
+        if self.yaml_params.is_nvme() or self.yaml_params.is_scm():
+            self.mode = "format"
+
+    def check_subprocess_status(self, subprocess):
+        """Wait for message from command output."""
+        patterns = {
+            "format": "SCM format required",
+            "normal": "I/O server instance.*storage ready",
+        }
+        start_time = time.time()
+        start_msgs = 0
+        timed_out = False
+        while start_msgs != self.server_cnt and not timed_out:
+            output = subprocess.get_stdout()
+            start_msgs = len(re.findall(patterns[self.mode], output))
+            timed_out = time.time() - start_time > self.timeout
+
+        if start_msgs != self.server_cnt:
+            err_msg = "{} detected. Only {}/{} messages received".format(
+                "Time out" if timed_out else "Error",
+                start_msgs, self.server_cnt)
+            self.log.info("{}:\n{}".format(err_msg, subprocess.get_stdout()))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments